### ISSUE TITLE
[Fixing bug] hide dropdown while autocompleting a non whitelisted text

### ISSUE THREAD

**Issue by mohamedaymenkarmous** [yairEO/tagify#171]
Hello,

This is my third issue opened today. I hope you are not tired of reading them.

I want to report and fix a bug while autocompleting a text that is not whitelisted.

This is a very specific bug. The bug appears only when the autocomplete is performed on a whitelisted text in the begining.

Example:

input="" --> dropdown=hide

input="\@" --> dropdown=hide

input="\@H" --> dropdown=show && suggestedListItems=[{value:"Homer"}]

input="\@Homer" --> dropdown=show && suggestedListItems=[{value:"Homer"}]

input="\@Homer2" --> dropdown=show && suggestedListItems=[{value:"Homer"}]

In the last step dropdown should be hidden && suggestedListItems=[] because Homer is whitelisted but since Homer2 is not whitelisted, the dropdown should be hidden

The fixing code is concerning [jQuery.tagify.min.js](https://github.com/yairEO/tagify/blob/9e2436fc7a3ecf2909bfb2b89f47e23b5926d979/dist/jQuery.tagify.min.js) file in dropdown.show method:

This is the current method:

```
            show: function(t) {
                var e, i = this;
                this.settings.whitelist.length && (this.suggestedListItems = t ? this.dropdown.filterListItems.call(this, t) : this.settings.whitelist.filter(function(t) {
                    return -1 == i.isTagDuplicate(t.value || t)
                }), e = this.dropdown.createListHTML.call(this, this.suggestedListItems), this.settings.autoComplete && this.input.autocomplete.suggest.call(this, this.suggestedListItems.length ? this.suggestedListItems[0].value : ""), e && (this.DOM.dropdown.innerHTML = e, this.dropdown.position.call(this), !this.DOM.dropdown.parentNode != document.body && (document.body.appendChild(this.DOM.dropdown), this.events.binding.call(this, !1), this.dropdown.events.binding.call(this))))
            },
```

The fixed method should be:

```
            show: function(t) {
                var e, i = this;
                this.settings.whitelist.length && (this.suggestedListItems = t ? this.dropdown.filterListItems.call(this, t) : this.settings.whitelist.filter(function(t) {
                    return -1 == i.isTagDuplicate(t.value || t)
                }), e = this.dropdown.createListHTML.call(this, this.suggestedListItems), this.settings.autoComplete && this.input.autocomplete.suggest.call(this, this.suggestedListItems.length ? this.suggestedListItems[0].value : ""), e ? (this.DOM.dropdown.innerHTML = e, this.dropdown.position.call(this), !this.DOM.dropdown.parentNode != document.body && (document.body.appendChild(this.DOM.dropdown), this.events.binding.call(this, !1), this.dropdown.events.binding.call(this))):this.dropdown.hide.call(this))
            },
```

What have I changed ?

In this part : e && (...)

If e (which is suggestedListItems) is found, we should render the HTML dropdown menu.

But if e is not found ? That's the problem. We should consider the situation where e is not found. And in that case we should hide the dropdown menu (I think if it's already opened if you want but I don't think that will make problem if we always hide it unless there is some logics behind it).

So I changed that part like this: e ? (...) : this.dropdown.hide.call(this)

I didn't want to create a pull request because I only corrected this issue in jQuery file and since the jQuery file is in min format It will be very difficult to review my code that's why I shared the fixing code here and I will let you the care to change it as you want.

I really liked how the code is optimized and I this helped you to fix it.

Thank you for your support

Best regards

**Comment by yairEO** [comment]
Don't worry, I encourage people to open issues! 
Please open as many as you wish, I want to improve the code and the issues are helping.

Btw, you pasted code from the `/dist` folder which is minified. The source code is in `/src/tagify.js`

**Comment by yairEO** [comment]
I will fix this ASAP. will hide the dropdown when there's no match with the whitelist, good idea

**Comment by mohamedaymenkarmous** [comment]
Hi

You're right. It's approximatively the same code. I saw that package.json is using /dist/tagify.min.js file and since /src/tagify.js is written differently I though that the /dist file can't be converted from the /src/tagify.js because there is no converted that can rewrite the functions like this and optimize all the instructions like this (only minify option is available in online converters).

Anyway.

Thank you again.

Best regards

**Comment by yairEO** [comment]
Related - 173

**Comment by yairEO** [comment]
Fixed in [2.10.1](https://github.com/yairEO/tagify/blob/2.10.0/src/tagify.js#L1014)
